### PR TITLE
for #10568 Make Search Fragment xml include toolbar and AwesomeBar

### PR DESCRIPTION
**Pull request by sraturi**
### Pull Request checklist

- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Screenshots**: This PR includes screenshots or GIFs of the changes made or an explanation of why it does not
- [ ] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/master/android/accessibility_guide.md) or does not include any user facing features. In addition, it includes a screenshot of a successful [accessibility scan](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor&hl=en_US) to ensure no new defects are added to the product.

### After merge
- [ ] **Milestone**: Make sure issues finished by this pull request are added to the [milestone](https://github.com/mozilla-mobile/fenix/milestones) of the version currently in development.

### To download an APK when reviewing a PR:
1. click on Show All Checks,
2. click Details next to "Taskcluster (pull_request)" after it appears and then finishes with a green checkmark,
3. click on the "Fenix - assemble" task, then click "Run Artifacts".
4. the APK links should be on the left side of the screen, named for each CPU architecture

### CODE REVIEW

**Comment by sraturi** [comment]
**Run 1** is a fresh install. **Run 2** is stopping the application and restarting in 10 seconds.

**PROFILES**
Run 1 with changes: https://share.firefox.dev/37E8Tb2
Run 2 with changes https://share.firefox.dev/30VHpMw

Run 1 Master: https://share.firefox.dev/3ehngV9
Run 2 Master: https://share.firefox.dev/3fBihij

**STATS**


  | Run 1 (Master) | Run 1 (This Branch) | Run 2 (Master) | Run 2 (This Branch)
-- | -- | -- | -- | --
Inflating search frag | 30, 42,43 = 38.3333 | 120,142,101= 121 | 26,27,28 = 27 | 88,76,88 = 84
Init awesome bar | 12, 14,13 = 13 | 1, 1,1=1 | 12,14,16=14 | 1, 1,1 = 1
Init toolbar | 69, 84,85 = 79.33 | 0 0,1= .333 | 44, 68,59 = 57 | 0,0,0
total(of the avg) | 130.666 | 122.333 | 98 | 85
  |   |   |   |  


**TLDR:  improvement**

Fresh install inflation time = 130.666 (Master) - 122.333(Changes) =  inflation time reduce by 8.336 MS

Second Run inflation time = 98 (Master) - 85(Changes) = inflation time reduced by 13 MS

File: app/src/main/java/org/mozilla/fenix/search/awesomebar/AwesomeBarView.kt [hunk 1]
```diff
@@ -82,14 +79,9 @@ interface AwesomeBarInteractor {
  */
 class AwesomeBarView(
     private val container: ViewGroup,
-    val interactor: AwesomeBarInteractor
-) : LayoutContainer {
-    val view: BrowserAwesomeBar = LayoutInflater.from(container.context)
```
Comment (sblatz) [inline app/src/main/java/org/mozilla/fenix/search/awesomebar/AwesomeBarView.kt:87]: As discussed, we don't need this any more because we have this logic [here](https://github.com/mozilla-mobile/fenix/blob/master/app/src/main/java/org/mozilla/fenix/components/toolbar/BrowserToolbarView.kt#L71-L75)
